From 168ee2e02c51ac4dd56621e7552f27526bc8d911 Mon Sep 17 00:00:00 2001
From: Norbert Preining <norbert@preining.info>
Date: Tue, 8 Mar 2016 22:55:27 +0900
Subject: [PATCH 25/25] adjust gpg code to spaces in filenames and win32

---
 Master/tlpkg/TeXLive/TLUtils.pm | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/Master/tlpkg/TeXLive/TLUtils.pm b/Master/tlpkg/TeXLive/TLUtils.pm
index b255588..b162b09 100644
--- a/Master/tlpkg/TeXLive/TLUtils.pm
+++ b/Master/tlpkg/TeXLive/TLUtils.pm
@@ -3638,12 +3638,12 @@ sub setup_gpg {
   # ok, we found one
   # Set up the gpg invocation:
   $::gpg = "$::progs{'gpg'} ";
-  if ($ENV{'TL_GNUPGHOME'}) {
-    # the user wants us to use this directory, sure, do it
-    $::gpg .= "--homedir $ENV{'TL_GNUPGHOME'} ";
-  } else {
-    $::gpg .= "--homedir $master/tlpkg/gpg ";
-  }
+  my $gpghome = ($ENV{'TL_GNUPGHOME'} ? $ENV{'TL_GNUPGHOME'} : 
+                                        "$master/tlpkg/gpg" );
+  $gpghome =~ s!/!\\!g if win32();
+  my $gpghome_quote = "\"$gpghome\"";
+  # mind the final space for following args
+  $::gpg .= "--homedir $gpghome_quote ";
   if ($ENV{'TL_GNUPGARGS'}) {
     $::gpg .= $ENV{'TL_GNUPGARGS'};
   } else {
@@ -3783,7 +3783,14 @@ GPGERROR
 
 sub gpg_verify_signature {
   my ($file, $sig) = @_;
-  my ($out, $ret) = run_cmd("$::gpg --verify $sig $file 2>&1");
+  my ($file_quote, $sig_quote);
+  if (win32()) {
+    $file =~ s!/!\\!g;
+    $sig =~ s!/!\\!g;
+  }
+  $file_quote = "\"$file\"";
+  $sig_quote = "\"$sig\"";
+  my ($out, $ret) = run_cmd("$::gpg --verify $sig_quote $file_quote 2>&1");
   if ($ret == 0) {
     debug("verification succeeded, output:\n$out\n");
     return 1;
-- 
2.7.0

