From 50154012a5959f553d7d581d3438398005dcdbb6 Mon Sep 17 00:00:00 2001
From: Norbert Preining <norbert@preining.info>
Date: Sat, 9 Apr 2016 13:58:51 +0900
Subject: [PATCH 34/39] fix missing access functions for new checksum

---
 Master/tlpkg/TeXLive/TLPOBJ.pm | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/Master/tlpkg/TeXLive/TLPOBJ.pm b/Master/tlpkg/TeXLive/TLPOBJ.pm
index 365aeed..bbed060 100644
--- a/Master/tlpkg/TeXLive/TLPOBJ.pm
+++ b/Master/tlpkg/TeXLive/TLPOBJ.pm
@@ -1209,6 +1209,21 @@ sub doccontainermd5 {
   if (@_) { $self->{'doccontainermd5'} = shift }
   return ( defined($self->{'doccontainermd5'}) ? $self->{'doccontainermd5'} : "" );
 }
+sub containerchecksum {
+  my $self = shift;
+  if (@_) { $self->{'containerchecksum'} = shift }
+  return ( defined($self->{'containerchecksum'}) ? $self->{'containerchecksum'} : "" );
+}
+sub srccontainerchecksum {
+  my $self = shift;
+  if (@_) { $self->{'srccontainerchecksum'} = shift }
+  return ( defined($self->{'srccontainerchecksum'}) ? $self->{'srccontainerchecksum'} : "" );
+}
+sub doccontainerchecksum {
+  my $self = shift;
+  if (@_) { $self->{'doccontainerchecksum'} = shift }
+  return ( defined($self->{'doccontainerchecksum'}) ? $self->{'doccontainerchecksum'} : "" );
+}
 sub srcsize {
   my $self = shift;
   if (@_) { $self->{'srcsize'} = shift }
-- 
2.8.0.rc3

